blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/MaxValue.java

```diff
@@ -100,17 +100,17 @@ void handleRollupMetric(IBasicRollup basicRollup) throws RuntimeException {
         }
 
         if (this.isFloatingPoint() && !other.isFloatingPoint()) {
-            if (this.toDouble() < (double)other.toDouble()) {
+            if (this.toDouble() < (double)other.toLong()) {
```

Comment:
Ah, gotcha, I neglected to see the first if statement:
```
if (this.isFloatingPoint() && !other.isFloatingPoint()) {
```
It makes sense now.

No need to revert. Fixing it was the right thing to do. I just didn't see it at first :-)

I agree about creating a new story to look into this later.